.github/workflows/static-checks-self-hosted.yaml

```diff
@@ -0,0 +1,130 @@
+on:
+  pull_request_target:
```

Comment:
hi @BbolroC , first of all, this is a critical problem so thanks for raising it (and fixing it!).

The static-check jobs don't rely on secrets so there isn't need to run on `pull_request_target`, we can run on `pull_request` and avoid all the nightmares bought by the former event.

Reply:
Oh, I didn't know that. Thanks for the info. I will change it back to `pull_request` again.